### ISSUE TITLE
How to train LSTM autoencoder for long sequences? Does not learn.

### ISSUE THREAD

**Issue by vivianta** [keras-team/keras#12489]
Hello every one. I am trying to train a LSTM autoencoder to echo a sequence of length 1500, which contains 50 tokens. My purpose is to use the output of the encoder layer as input to other network. Thinking sequence of length 1500 could be too long, I started with sequence of length 100, and the model is as below.

To train the network, I use random genered sequence of the 50 tokens as input data. However, after almost 10 hours training my network does not learn, loss not decreasing and accuracy not increasing.

My question is 1. Is the number of cells in my encoder and decoder layer suitable for my sequence length? and 2. How should I train my network effectively?

Thank you for kind help!

`encoder_cell=200
decoder_cell=200
n_in=100
n_out=100
code_len=50 # one hot encoded input of 50 tokens

model = Sequential()
model.add(LSTM(encoder_cell, input_shape=(n_in, code_len), stateful=False)) # From 4.3 on
model.add(RepeatVector(n_out))
model.add(LSTM(decoder_cell, return_sequences=True, stateful=False))
model.add(TimeDistributed(Dense(code_len, activation='softmax')))
model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['acc'])
`

**Comment by vivianta** [comment]
encoder_cell=200
decoder_cell=200
n_in=100
n_out=100
code_len=50 # one hot encoded input of 50 tokens
model = Sequential()
    model.add(LSTM(encoder_cell, input_shape=(n_in, code_len), stateful=False)) # From 4.3 on
    model.add(RepeatVector(n_out))
    model.add(LSTM(decoder_cell, return_sequences=True, stateful=False))
    model.add(TimeDistributed(Dense(code_len, activation='softmax'))) 
    model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['acc'])
